src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs

```diff
@@ -2195,13 +2166,16 @@ internal unsafe void UpdateImage()
         TVITEMW item = new()
         {
             mask = TVITEM_MASK.TVIF_HANDLE | TVITEM_MASK.TVIF_IMAGE,
-            hItem = Handle,
+            hItem = HTREEITEMInternal,
             iImage = Math.Max(0, ((ImageIndexer.ActualIndex >= tv.ImageList.Images.Count) ? tv.ImageList.Images.Count - 1 : ImageIndexer.ActualIndex))
         };
 
         PInvoke.SendMessage(tv, PInvoke.TVM_SETITEMW, 0, ref item);
     }
 
+    HTREEITEM IHandle<HTREEITEM>.Handle => HTREEITEMInternal;
+    internal HTREEITEM HTREEITEMInternal => (HTREEITEM)_handle;
```

Comment:
Digging through the code again, I don't see anywhere we are doing P/Invoke calls that explicitly need HTREEITEM, so I am not sure that implementing `IHandle<HTREEITEM>` is totally necessary. Let's remove the implementation for now and we can easily add it if we see it is needed later, my apologies for that. 🙏 
We should still add some internal properties for getting the HTREEITEM though, so this is close to what we want, but as it is right now, this would be changing the behavior since we are no longer calling `Handle`, which would force handle creation if it hasn't already been created. I think we should have something like the following:
```suggestion
    internal HTREEITEM HTREEITEM => (HTREEITEM)Handle
    internal HTREEITEM HTREEITEMInternal => (HTREEITEM)_handle;
```

then anywhere that was originally calling `Handle` can be replaced with `HTREEITEM` (I believe I also saw a few in TreeNodeTest.cs) while anywhere that was originally calling `_handle` can be replaced with `HTREEITEMInternal`. You may be able to actually get rid of `_handle` entirely and just have `HTREEITEMInternal` serve as an auto prop for it, but I am not entirely sure by just looking 🤔